pkg/agent/interfacestore/interface_cache.go

```diff
@@ -43,6 +43,8 @@ import (
 type interfaceCache struct {
 	sync.RWMutex
 	cache map[string]*InterfaceConfig
+	// Interfaces without IP are not stored in ipCache
+	ipCache map[string]*InterfaceConfig
```

Comment:
For your information, PR #827 introduces Indexer for the interface cache. You might rebase and switch to using that.

Reply:
Thanks for the suggestion. I rebased and used indexer to update the patch. Indexer seems to be a nice way to maintain a Store.